src/storage-client/src/client.rs

```diff
@@ -275,6 +275,21 @@ impl Arbitrary for StorageCommand<mz_repr::Timestamp> {
     }
 }
 
+/// This structure represents a full set up updates for the `mz_storage_source_statistics`
+/// table, for a specific source-worker pair. It is structured like this for simplicity
+/// and efficiency: Each storage worker can individually collect and consolidate metrics,
+/// then control how much `StorageResponse` traffic is produced when sending updates
+/// back to the controller to be written.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct SourceStatisticsUpdate {
```

Comment:
musing: should we maybe include the (wall-clock, I know ... 🙈) time at which the metrics were produced?

Reply:
I have a comment about this somewhere in the pr...

basically my conclusion is that we should just allow some jitter (note also that the `CollectionManager` we use in the controller to record these doesn't support setting a value at a ts, and adding that would be fairly complex), because this is a simple implementation that is a stopgap; when we move to using logging dataflows and doing this correctly™, we definitely want to be more specific about the times

Reply:
https://github.com/MaterializeInc/materialize/pull/16788/files#r1058531089